tests/common/devices.py

```diff
@@ -232,14 +232,20 @@ def get_crm_resources(self):
 
         return result
 
-    def get_pmon_daemon_list(self):
+    def get_pmon_daemon_states(self):
         """
-        @summary: get pmon daemon list from the config file (/usr/share/sonic/device/{platform}/{hwsku}/pmon_daemon_control.json)
+        @summary: get state list of daemons from pmon docker.
+                  Referencing (/usr/share/sonic/device/{platform}/{hwsku}/pmon_daemon_control.json)
                   if some daemon is disabled in the config file, then remove it from the daemon list.
+
+        @return: dictionary of { service_name1 : state1, ... ... }
         """
-        full_daemon_tup = ('xcvrd', 'ledd', 'psud', 'syseepromd')
+        # some services are meant to have a short life span or not part of the daemons
+        exemptions = ('lm-sensors', 'start.sh', 'rsyslogd')
+
+        daemons = self.shell('docker exec pmon supervisorctl status')['stdout_lines']
+
         daemon_ctl_key_prefix = 'skip_'
-        daemon_list = []
         daemon_config_file_path = os.path.join('/usr/share/sonic/device', self.facts["platform"], 'pmon_daemon_control.json')
```

Comment:
don't you need hwsku info? the path specified in the summary

Reply:
`pmon_daemon_control.json` is platform-specific, and is shared among that SKUs for that platform. Thus, it is under the platform directory (`self.facts["platform"]`).